gtsfm/averaging/rotation/shonan.py

```diff
@@ -100,6 +104,15 @@ def run(self, num_images: int, i2Ri1_dict: Dict[Tuple[int, int], Optional[Rot3]]
                 underconstrained system or ill-constrained system), or where the camera pose had no valid observation
                 in the input to run().
         """
+        if len(i2Ri1_dict) == 0:
+            logger.warning(
+                "Shonan cannot proceed: No cycle-consistent triplets found, "
+                + "so no measurements survived the filtering step."
+            )
+
```

Comment:
nit: I think a blank line here is unnecessary.

Reply:
thanks, should be fixed now